torch/_inductor/codegen/cpp_wrapper_gpu.py

```diff
@@ -409,12 +428,20 @@ def generate_kernel_call(
 
         call_args_str = self.generate_args_decl(call_args, arg_types)
         kernel_args_var = f"kernel_args_var_{next(self.kernel_callsite_id)}"
-        self.writeline(f"void* {kernel_args_var}[] = {{{call_args_str}}};")
+        if self.device == "xpu":
+            self.writeline(
+                f"std::vector<void*> {kernel_args_var} = {{{call_args_str}}};"
```

Comment:
Why there needs to be a difference for xpu here?

Reply:
Hi @desertfire, here I use vector to contain the number of kernel args, which will be used when launching kernel in SYCL runtime.